### PR TITLE
Improve error logging in std::Package

### LINKED ISSUE #238: Improved error reporting std::Package
**Problem:**

The handler for the `std::Package` discards error messages by providing the options `-d 0 -e 0` to dnf or yum. As such, error messages are not included in the resource log. 

https://github.com/inmanta/std/blob/2b14abca12634cddd647f1b8211adfbc51f537fe/plugins/resources.py#L463

One of the methods to reproduce this is to try to install a non-existant package like: abcdefg

### PR DESCRIPTION
**Pull request by jptrindade**
# Description

The handler for the std::Package discards error messages by providing the options `-d 0 -e 0` to dnf or yum. These flags have been deprecated and replaced with the `-v` flag (--verbose).

closes #238 

# Merge procedure

Don't use the github built-in merge, but the process described [here](https://internal.inmanta.com/developer/tasks/commiting_changes_modules.html)

```sh
git pull
git checkout master
git pull
git merge --squash issue/{issue-number}-{short description}
inmanta module commit -m "{Commit Message Here}" -r
git push
git push {tag} # push the tag as well
```

Then close the PR with a reference to the commit

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Version number is bumped to dev version
~~- [ ] Code is clear and sufficiently documented~~
~~- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)~~
~~- [ ] Correct, in line with design~~
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

### CODE REVIEW

**Comment by jptrindade** [comment]
The version was already on 4.0.2 but no Changelog entry. I think something is wrong.

**Comment by bartv** [comment]
We disabled all output. This caused all errors to be hidden, so for example the install with the non-existing package. The existing behavior was:

```
$ sudo dnf -d 0 -e 0 install sfdsdfs

```

You patch will replace this with:

```
$ sudo dnf -v install sfdsdfs
Loaded plugins: builddep, changelog, config-manager, copr, debug, debuginfo-install, download, generate_completion_cache, groups-manager, needs-restarting, playground, repoclosure, repodiff, repograph, repomanage, reposync, system-upgrade
DNF version: 4.14.0
cachedir: /var/cache/dnf
User-Agent: constructed: 'libdnf (Fedora Linux 37; workstation; Linux.x86_64)'
repo: using cache for: copr:copr.fedorainfracloud.org:jdoss:slack-repo
copr:copr.fedorainfracloud.org:jdoss:slack-repo: using metadata from Wed 10 Aug 2022 05:20:38 PM CEST.
repo: using cache for: copr:copr.fedorainfracloud.org:kwizart:fedy
copr:copr.fedorainfracloud.org:kwizart:fedy: using metadata from Tue 22 Nov 2022 04:28:07 AM CET.
repo: using cache for: copr:copr.fedorainfracloud.org:phracek:PyCharm
copr:copr.fedorainfracloud.org:phracek:PyCharm: using metadata from Fri 30 Dec 2022 12:01:04 PM CET.
repo: using cache for: azurecli
azurecli: using metadata from Tue 05 Jul 2022 05:02:20 AM CEST.
repo: using cache for: docker-ce-stable
docker-ce-stable: using metadata from Mon 19 Dec 2022 10:30:22 PM CET.
repo: using cache for: Dropbox
Dropbox: using metadata from Thu 05 Mar 2020 12:33:11 AM CET.
repo: using cache for: fedora
fedora: using metadata from Thu 10 Nov 2022 10:30:00 AM CET.
repo: using cache for: fedora-cisco-openh264
fedora-cisco-openh264: using metadata from Thu 06 Oct 2022 01:01:40 PM CEST.
repo: using cache for: fedora-modular
fedora-modular: using metadata from Thu 10 Nov 2022 10:23:24 AM CET.
repo: using cache for: updates
updates: using metadata from Fri 06 Jan 2023 02:38:01 AM CET.
repo: using cache for: updates-modular
updates-modular: using metadata from Tue 03 Jan 2023 02:27:52 AM CET.
repo: using cache for: google-chrome
google-chrome: using metadata from Thu 05 Jan 2023 04:19:40 PM CET.
repo: using cache for: packages.microsoft.com_yumrepos_ms-teams_
packages.microsoft.com_yumrepos_ms-teams_: using metadata from Mon 19 Sep 2022 01:43:00 PM CEST.
repo: using cache for: rpmfusion-free
rpmfusion-free: using metadata from Fri 11 Nov 2022 05:54:42 PM CET.
repo: using cache for: rpmfusion-free-updates
rpmfusion-free-updates: using metadata from Tue 03 Jan 2023 02:02:41 PM CET.
repo: using cache for: rpmfusion-nonfree
rpmfusion-nonfree: using metadata from Fri 11 Nov 2022 06:25:12 PM CET.
repo: using cache for: rpmfusion-nonfree-nvidia-driver
rpmfusion-nonfree-nvidia-driver: using metadata from Wed 21 Dec 2022 12:34:11 AM CET.
repo: using cache for: rpmfusion-nonfree-steam
rpmfusion-nonfree-steam: using metadata from Wed 24 Aug 2022 06:15:54 PM CEST.
repo: using cache for: rpmfusion-nonfree-updates
rpmfusion-nonfree-updates: using metadata from Tue 03 Jan 2023 02:27:44 PM CET.
repo: using cache for: teams
teams: using metadata from Mon 19 Sep 2022 01:43:00 PM CEST.
repo: using cache for: code
code: using metadata from Fri 06 Jan 2023 09:42:43 AM CET.
repo: using cache for: yum.fury.io_netdevops_
yum.fury.io_netdevops_: using metadata from Fri 23 Dec 2022 11:07:32 AM CET.
Last metadata expiration check: 1:35:07 ago on Fri 06 Jan 2023 03:27:06 PM CET.
No match for argument: sfdsdfs
Error: Unable to find a match: sfdsdfs
```

This is not much of an improvement. I did some quick testing and by setting "-e 1" we get a much better result:

```
$ sudo dnf -d 0 -e 1 install sfdsdfs
Error: Unable to find a match: sfdsdfs
```

**Comment by wouterdb** [comment]
@jptrindade the tests are failing on jenkins, I think (as I expected) that the changes on the output break the parsing. 

This definitely has to be addressed.

**Comment by jptrindade** [comment]
@bartv I changed `-d` and `-e` to `-v` because those options are deprecated and on the docs this was the suggested solution.
Unfortunately the `-v` does not have the same level of granularity when choosing what to output (at least it is not documented).
Do you suggest we keep using the `-d` and `-e`?

**Comment by bartv** [comment]
> @bartv I changed `-d` and `-e` to `-v` because those options are deprecated and on the docs this was the suggested solution. Unfortunately the `-v` does not have the same level of granularity when choosing what to output (at least it is not documented). Do you suggest we keep using the `-d` and `-e`?

Did you try to find a ticket that explains when these will be removed? What about other systems like ansible, how do they do it?

For sure the -v option is absolutely useless. Did you try it yourself?

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: plugins/resources.py [hunk 1]
```diff
@@ -460,9 +460,9 @@ def _parse_fields(self, lines):
     def _run_yum(self, args):
         # todo: cache value
         if self._io.file_exists("/usr/bin/dnf"):
-            return self._io.run("/usr/bin/dnf", ["-d", "0", "-e", "0", "-y"] + args)
+            return self._io.run("/usr/bin/dnf", ["-d", "0", "-e", "1", "-y"] + args)
```
[the remark this answers is not in the document]
Comment (bartv) [reply plugins/resources.py:463]: Lets add a test case because it is super easy to trigger. Just try to install a non existing package

**Review by wouterdb — APPROVED** [review]

**Review by bartv — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
-## v4.1.1 - ?
```
Comment (wouterdb) [inline CHANGELOG.md:3]: this `- ?` is important for the tooling to work correctly.

**Review by wouterdb — CHANGES_REQUESTED** [review]

**Review by wouterdb — APPROVED** [review]

**Pull request merged**